Model/Product/Index/IndexRepository.php

```diff
@@ -99,6 +99,29 @@ public function getById($id)
         return $collection->getFirstItem(); // @codingStandardsIgnoreLine
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function getTotalOutOfSync()
+    {
+        /* @var IndexCollection $collection */
+        $collection = $this->indexCollectionFactory->create();
+        return $collection
+            ->addFilter(ProductIndexInterface::IN_SYNC, NostoIndex::DB_VALUE_BOOLEAN_FALSE, 'eq')
+            ->getSize();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getTotalDirty()
```

Comment:
Add optional Store parameters. If Store is present limit the results only for that store.